src/main/java/task/Task.java

```diff
@@ -0,0 +1,54 @@
+package task;
+
+/**
+ * This class represents a task.
+ */
+public class Task {
+    protected String description;
+    protected boolean isCompleted;
+
+    /**
+     * Constructs a new Task object with the given description.
+     *
+     * @param description description of task.
+     * @param isCompleted true when tasks is completed.
+     */
+    public Task(String description, boolean isCompleted) {
+        this.description = description;
+        this.isCompleted = isCompleted;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public boolean isCompleted() {
+        return this.isCompleted;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void markComplete() {
+        this.isCompleted = true;
+    }
+
+    /**
+     * Marks the class as incompleted.
+     */
+    public void markIncomplete() {
+        this.isCompleted = false;
+    }
+
+    /**
+     * Returns the String format of how tasks are listed in the list of task.
+     *
+     * @return User-readable task in String format
+     */
+    public String toString() {
+        if (this.isCompleted) {
```

Comment:
Not a coding standard violation and this is fine, but can consider having a `completionIcon()` method which returns "[ ]" or "[X]" depending on the boolean isCompleted status instead of statically using if & else to determine which one to print

then this method can just be `return this.completionIcon() + " " + this.description`

will be neater if you decide to add more components to this `toString()` method in the future as you have more control over the individual components :)

Reply:
Thank you for your advice! It really does seem neater if I implement it that way. I will try :)